Sources/LeafKit/LeafRenderer.swift

```diff
@@ -21,6 +21,21 @@ final class Cache: LeafCache {
     }
 }
 
+protocol CustomTagProtocol {
+    func render(params: [ProcessedParameter],
+                   body: [Syntax]?,
+                   context: [String: TemplateData]) throws -> TemplateData
```

Comment:
If we call the tag type `LeafTag`, it would be nice to rename `TemplateData` -> `LeafData`. Going with `Leaf...` might make more sense since we eventually want to add packages for things like Mustache. If both of those declare `TemplateData`, that could become a mess.